backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/ResilientClusterProvider.java

```diff
@@ -63,6 +67,7 @@ private ResilientClusterProvider(ClusterConfiguration configuration) {
             Cluster cluster = ClusterFactory.create(configuration);
             try {
                 KeyspaceFactory.createKeyspace(configuration, cluster);
+                ensureContactable(cluster);
```

Comment:
Can you explain to me:
 - 1. why we ensure contactable after creating the keyspace?
 - 2. Why we need to ensure the cluster is contactable, why it was working before and why this change is needed...

Reply:
> why it was working before and

It wasn't. We just missed a test case. the `Cluster cluster = ClusterFactory.create(configuration);` doesn't thrown regardless the status of cassandra (I paused). So if I don't execute any query in `KeyspaceFactory` by setting `configuration.createKeyspace` to false, there's no way to know my cluster can connect to caassandra and execute queries or not. 

That's the reason I need to query something with the cluster.

Reply:
Shouldn't it be done and handled in KeyspaceFactory then?

Reply:
did you mean `ClusterFactory` ? I don't think that is the responsibility of `KeyspaceFactory`

Reply:
Even better ;-)